src/pages/index.js

```diff
@@ -0,0 +1,304 @@
+let selectedCard;
+let selectedCardId;
+
+//Functions
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  function handleLike(evt, id) {
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_liked");
+    api
+      .changeLikeStatus(data._id, isLiked)
+      .then((res) => {
+        cardLikeBtn.classList.toggle("card__like-btn_liked", res.isLiked);
+      })
+      .catch(console.error);
+  }
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  cardNameEl.textContent = data.name;
+
+  cardImageEl.src = data.link;
+
+  cardImageEl.alt = data.name;
+
+  return cardElement;
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEsc);
+  modal.removeEventListener("mousedown", closeOverlay);
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEsc);
+  modal.addEventListener("mousedown", closeOverlay);
+}
+
+function closeOverlay(e) {
+  if (e.target.classList.contains("modal")) {
+    closeModal(e.target);
+  }
+}
+
+function closeModalEsc(e) {
+  if (e.key === "Escape") {
+    const modalOpened = document.querySelector(".modal_opened");
+    closeModal(modalOpened);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  const inputValue = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addNewCard(inputValue)
+    .then((res) => {
+      renderCard(res);
+      closeModal(cardModal);
+      evt.target.reset();
+      disabledButton(avatarSubmitBtn, settings);
```

Comment:
`avatarSubmitBtn` is for the avatar form

* According to the previous task you need to **reset** the form and **disable** the save button in the `card` form after submission https://snipboard.io/fjmNYG.jpg  . So, in the 9th sprint the save button should be also disabled if the form is invalid after submission (inputs are empty) https://snipboard.io/OSLjsF.jpg 


You have `submitBtn` in line 184 as the submit button of the current form


```js
      disabledButton(submitBtn, settings);


```